IntelliTect.TestTools.Selenate/IntelliTect.TestTools.Selenate/WebElement.cs

```diff
@@ -0,0 +1,274 @@
+
+            return
+                new ReadOnlyCollection<WebElement>(
+                    WrappedElement.FindElements(by)
+                        .Select(webElement =>
+                        new WebElement(WrappedElement.FindElement(by), by, _Driver))
+                        .ToList());
+        }
+
+        void IWebElement.Clear()
+        {
+            Clear().GetAwaiter().GetResult();
+        }
+        public Task Clear()
+        {
+            return RetryAction("clear");
+        }
+
+        void IWebElement.SendKeys(string text)
+        {
+            SendKeys(text).ConfigureAwait(false).GetAwaiter().GetResult();
+        }
+
+        public Task SendKeys(string text)
+        {
+            return RetryAction("sendkeys", text);
+        }
+
+        void IWebElement.Submit()
+        {
+            Submit().GetAwaiter().GetResult();
+        }
+
+        public Task Submit()
+        {
+            return RetryAction("submit");
+        }
+
+        void IWebElement.Click()
+        {
+            Click().GetAwaiter().GetResult();
+        }
+
+        public Task Click()
+        {
+            return RetryAction("click");
+        }
+
+        // Can this easily be abstracted out to an extension method?
+        // Seems to make more sense as an extension
+        public void Click(int timeToRetry)
+        {
+            TimeToRetry = timeToRetry;
+            Click();
+            TimeToRetry = DefaultRetrySeconds;
+        }
+
+        string IWebElement.GetAttribute(string attributeName)
+        {
+            return GetAttribute(attributeName).GetAwaiter().GetResult();
+        }
+
+        public Task<string> GetAttribute(string attributeName)
+        {
+            return RetryAction("getattribute", attributeName);
+        }
+
+        string IWebElement.GetProperty(string propertyName)
+        {
+            return GetProperty(propertyName).GetAwaiter().GetResult();
+        }
+
+        public Task<string> GetProperty(string propertyName)
+        {
+            return RetryAction("getproperty", propertyName);
+        }
+
+        string IWebElement.GetCssValue(string propertyName)
+        {
+            return GetCssValue(propertyName).GetAwaiter().GetResult();
+        }
+
+        public Task<string> GetCssValue(string propertyName)
+        {
+            return RetryAction("getcss", propertyName);
+        }
+
+        private const int DefaultRetrySeconds = 30;
+        private int TimeToRetry { get; set; } = DefaultRetrySeconds;
+        private IWebElement WrappedElement { get; set; }
+        private readonly IWebDriver _Driver;
+
+        private async Task<string> RetryAction(string action, string text = null)
+        {
+            DateTime end = DateTime.Now.AddSeconds(TimeToRetry);
+            bool reFindElement = !Initialized;
+            List<Exception> retryExceptions = new List<Exception>();
+            while (DateTime.Now <= end)
+            {
+                await Task.Delay(250);
```

Comment:
Again generally you put the delay after the action. That way if it can be completed on the first try the caller isn't forced to wait the 250 milliseconds.